cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -103,12 +105,8 @@ public BasicMapReduceContext(Program program,
           getMetricsCollector(program, runId.getId(), metricsCollectionService),
           dsFramework, txClient, discoveryServiceClient, false, pluginInstantiator);
     this.workflowProgramInfo = workflowProgramInfo;
-
-    if (metricsCollectionService != null) {
-      this.userMetrics = new ProgramUserMetrics(getProgramMetrics());
-    } else {
-      this.userMetrics = null;
-    }
+    this.programMetricsContext = getProgramMetrics();
+    this.workFlowMetricsContext = getWorkflowMetricsContext(workflowProgramInfo, program, metricsCollectionService);
```

Comment:
i initially did that, then figured its better to do that in the method and have `Nullable` annotation.